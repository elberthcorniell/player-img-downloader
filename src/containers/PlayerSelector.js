import DropdownSearch from "../components/Dropdown";
import Download from "../components/Download";
import { useState, useRef } from 'react';
import sportArray from '../data/sport.json';
import countryArray from '../data/country.json';
import playerArray from '../data/player.json';
import {
    Row,
    Col,
    Card,
} from "react-bootstrap";
import {
    rowMargin,
    cardBody,
    downloadButton,
} from "../styles/style";
import { handleImgError } from '../utils';

const PlayerSelector = () => {
    const [player, setPlayer] = useState({});
    const [sport, setSport] = useState({});
    const [country, setCountry] = useState({});
    const imgRef = useRef(null);

    return <Row style={rowMargin}>
        <Col lg={{ span: 8, offset: 2 }} sm={12}>
            <Card>
                <Card.Body style={cardBody}>
                    <Row style={rowMargin}>
                        <Col lg={4} sm={12}>
                            <DropdownSearch
                                buttonText={sport.name || 'Select Sport'}
                                items={sportArray}
                                onClick={setSport}
                            />
                        </Col>
                        <Col lg={4} sm={12}>
                            <DropdownSearch
                                buttonText={country.name || 'Select Country'}
                                items={countryArray}
                                onClick={setCountry}
                                disabled={!sport.name}
                            />
                        </Col>
                        <Col lg={4} sm={12}>
                            <DropdownSearch
                                buttonText={player.name || 'Select Player'}
                                items={playerArray?.filter(x => {
                                    return x.country_code === country.country_code && x.sport_code === sport.sport_code
                                }
                                )}
                                onClick={setPlayer}
                                disabled={!sport.name || !country.name}
                            />
                        </Col>
                    </Row>
                    <h1>{player.name || 'Select a Player'}</h1>
                    <img
                        src={player.url_image || '/'}
                        ref={imgRef}
                        alt={player.name}
                        style={{
                            maxWidth: '100%',
                            maxHeight: 400,
                        }}
                        onError={() => handleImgError(imgRef)}
                    />
                    <br />
                    {player.name && <Download
                        style={downloadButton}
                        url={player.url_image}
                    >
                        Download
                    </Download>}
                </Card.Body>
            </Card>
        </Col>
    </Row >
}

export default PlayerSelector;
