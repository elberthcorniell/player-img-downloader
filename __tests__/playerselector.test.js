import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import PlayerSelector from '../src/containers/PlayerSelector';

describe('Player Selector Container', () => {
    test('Render as expected', () => {
        render(<PlayerSelector />)
        expect(document.body.innerHTML).toMatchSnapshot();
    });
});