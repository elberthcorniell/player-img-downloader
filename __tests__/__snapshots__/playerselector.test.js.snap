// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player Selector Container Render as expected 1`] = `"<div><div style=\\"margin: 20px 0px;\\" class=\\"row\\"><div class=\\"col-lg-8 col-sm-12 offset-lg-2\\"><div class=\\"card\\"><div class=\\"card-body\\" style=\\"text-align: center; position: relative; padding-bottom: 80px;\\"><div style=\\"margin: 20px 0px;\\" class=\\"row\\"><div class=\\"col-lg-4 col-sm-12\\"><div class=\\"dropdown\\"><button style=\\"width: 100%; margin-bottom: 10px;\\" type=\\"button\\" class=\\"btn btn-primary\\">Select Sport ▼</button></div></div><div class=\\"col-lg-4 col-sm-12\\"><div class=\\"dropdown\\"><button style=\\"width: 100%; margin-bottom: 10px;\\" type=\\"button\\" class=\\"btn btn-primary\\">Select Country ▼</button></div></div><div class=\\"col-lg-4 col-sm-12\\"><div class=\\"dropdown\\"><button style=\\"width: 100%; margin-bottom: 10px;\\" type=\\"button\\" class=\\"btn btn-primary\\">Select Player ▼</button></div></div></div><h1>Select a Player</h1><img src=\\"/\\" style=\\"max-width: 100%; max-height: 400px;\\"><br></div></div></div></div></div>"`;
